### PR TITLE
New Plugin Architecture

### PR DESCRIPTION
**Pull request by christophwitzko**
#379

### CODE REVIEW

**Comment by Charlotteis** [comment]
Hai @christophwitzko! Great work as always.

I have a request:

On several occasions you do an `if (thing) do thing`
I was wondering if we could be a little more consistent and accessible by sticking to the more verbose

```
if (thing) {
    do thing
}
```

It may seem a little silly at times, but I feel like this is a more readable way of doing things.
This has come up before with others and I think we agreed on this way in the past, but I'm happy to discuss it further as always.

:ok_hand:

**Comment by gr2m** [comment]
looks good to me :+1:

File: lib/hapi/api/internals.js [hunk 1]
```diff
@@ -72,13 +72,11 @@ module.exports = {
...
     var pluginName = request.params.name
-    if (!pluginName) {
-      notFound(request, reply)
-      return
-    }
+    if (!pluginName) return notFound(request, reply)
 
     // the plugin is now responsible to call `reply()`
-    hooks.runDynamicForPlugin(pluginName, 'server.api.plugin-request', [request, reply])
+    var eres = hooks.emit(pluginName.replace(':', '-') + ':request', request, reply)
```
Comment (boennemann) [inline lib/hapi/api/internals.js]: can the colon appear more than once? `replace(/:/g, '-')`

**Comment by boennemann** [comment]
I merged this onto next now, which released [12.0.0](https://github.com/hoodiehq/hoodie-server/releases/tag/v12.0.0).

This evolved into a very long running monster, which got us into lots of (greenkeeper) merge conflicts.
I'll work on smaller feature branches now, plus these changes require some changes in the core plugins and start script, so once these are all out we can cut another top-level hoodie module release.

**Pull request closed without merging**